### PR TITLE
Add check for empty object string in claims request parameter

### PR DESCRIPTION
**Pull request by pkanher617**
This PR checks whether the object passed to the claims parameter is not an empty object. This prevents unnecessary cache skips when the empty object is provided, as we are only required to skip if additional claims are requested.

### CODE REVIEW

File: lib/msal-common/src/utils/Constants.ts [hunk 1]
```diff
@@ -38,7 +38,8 @@ export const Constants = {
     AUTHORIZATION_PENDING: "authorization_pending",
     NOT_DEFINED: "not_defined",
     EMPTY_STRING: "",
-    FORWARD_SLASH: "/"
+    FORWARD_SLASH: "/",
+    EMPTY_OBJECT_STRING: "{}"
```
Comment (tnorling) [inline lib/msal-common/src/utils/Constants.ts:42]: What if someone passes "{ }"? Would it be better to do regex matching? Or at least remove whitespace before comparison?

Comment (sameerag) [reply lib/msal-common/src/utils/Constants.ts:42]: +1 to regex matching

File: lib/msal-common/test/account/AuthToken.spec.ts [hunk 2]
```diff
@@ -77,8 +77,9 @@ describe("AuthToken.ts Class Unit Tests", () => {
         it("Throws error if rawIdToken is null or empty", () => {
             expect(() => new AuthToken("", cryptoInterface)).to.throw(ClientAuthErrorMessage.nullOrEmptyToken.desc);
             expect(() => new AuthToken("", cryptoInterface)).to.throw(ClientAuthError);
-
+            // @ts-ignore
```
Comment (sameerag) [inline lib/msal-common/test/account/AuthToken.spec.ts:80]: exception to function params being non-nullable?

Comment (sameerag) [reply lib/msal-common/test/account/AuthToken.spec.ts:80]: Can we just add an exception to tests?

Comment (pkanher617) [reply lib/msal-common/test/account/AuthToken.spec.ts:80]: this is more to test vanilla JS scenarios. Typescript will catch them regardless. We are making some changes to our tests, one of them may be to test vanilla JS and typescript separately.

File: lib/msal-common/src/utils/Constants.ts (hunk 1 above)
Comment (pkanher617) [reply lib/msal-common/src/utils/Constants.ts:42]: Good point, I will make some changes.

File: lib/msal-common/test/account/AuthToken.spec.ts (hunk 2 above)
Comment (pkanher617) [reply lib/msal-common/test/account/AuthToken.spec.ts:80]: > Can we just add an exception to tests?

Not sure what you mean.

Comment (sameerag) [reply lib/msal-common/test/account/AuthToken.spec.ts:80]: Allow null in function params for test files in lint rules.

File: lib/msal-common/src/utils/Constants.ts (hunk 1 above)
Comment (jasonnutter) [reply lib/msal-common/src/utils/Constants.ts:42]: Could consider using `JSON.parse` here also.

Comment (jasonnutter) [reply lib/msal-common/src/utils/Constants.ts:42]: e.g. `Object.keys(JSON.parse(" {  } ")).length`

**Review by tnorling — APPROVED** [review]

File: lib/msal-common/src/utils/Constants.ts (hunk 1 above)
Comment (pkanher617) [reply lib/msal-common/src/utils/Constants.ts]: done.

File: lib/msal-common/src/utils/StringUtils.ts [hunk 3]
```diff
@@ -42,6 +42,20 @@ export class StringUtils {
...
+     * @param strObj 
+     */
+    static isEmptyObj(strObj?: string): boolean {
+        if (strObj && !StringUtils.isEmpty(strObj)) {
+            try {
+                const obj = JSON.parse(strObj);
+                return Object.keys(obj).length === 0;
+            } catch (e) {}
+        }
+        return true;
```
Comment (jasonnutter) [inline lib/msal-common/src/utils/StringUtils.ts:56]: Do we want to return true or false here? Seems like `false` should be teh default?

**Review by jasonnutter — APPROVED** [review]

**Pull request merged**